### PR TITLE
[SPARK-48253][PS][SQL] Support default mode for Pandas API on Spark

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to support default mode for Pandas API on Spark.

### Why are the changes needed?

To provide option to users for reducing the communication cost between Python process and JVM when they don't want to manage the pandas-on-Spark options.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

```shell
./bin/pyspark --conf "spark.sql.pyspark.pandas.defaultMode=true"
```

```python
>>> import pyspark.pandas as ps
>>> ps.set_option("plotting.max_rows", 10)
>>> ps.get_option("plotting.max_rows")
'1000'
```

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by itholic** [comment]
cc @HyukjinKwon

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/pandas/config.py [hunk 1]
```diff
@@ -396,8 +398,8 @@ def set_option(key: str, value: Any) -> None:
     _check_option(key)
     _options_dict[key].validate(value)
     spark_session = default_session()
-
-    spark_session.conf.set(_key_format(key), json.dumps(value))
+    if not get_default_mode():
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply python/pyspark/pandas/config.py:401]: Yeah let's add

[the remark this answers is not in the document]
Comment (itholic) [reply python/pyspark/pandas/config.py:402]: Sounds good. Just added

File: python/pyspark/pandas/config.py [hunk 2]
```diff
@@ -396,8 +399,12 @@ def set_option(key: str, value: Any) -> None:
...
+        spark_session.conf.set(_key_format(key), json.dumps(value))
+    warnings.warn(
```
Comment (gatorsmile) [inline python/pyspark/pandas/config.py:404]: Add Else?

File: python/pyspark/pandas/config.py [hunk 3]
```diff
@@ -374,8 +375,10 @@ def get_option(key: str, default: Union[Any, _NoValueType] = _NoValue) -> Any:
         default = _options_dict[key].default
     _options_dict[key].validate(default)
     spark_session = default_session()
-
-    return json.loads(spark_session.conf.get(_key_format(key), default=json.dumps(default)))
+    if get_default_mode():
+        return json.dumps(default)
```
Comment (ueshin) [inline python/pyspark/pandas/config.py:379]: We don't need `json.dumps` here?

File: python/pyspark/pandas/utils.py [hunk 4]
```diff
@@ -492,6 +504,10 @@ def default_session() -> SparkSession:
             "from pandas API on Spark since pandas API on Spark follows "
             "the behavior of pandas, not SQL."
         )
+    default_mode_value = get_default_mode()
+    if default_mode_value is None:
+        default_mode_from_conf = spark.conf.get("spark.sql.pyspark.pandas.defaultMode") == "true"
```
[the remark this answers is not in the document]
Comment (itholic) [reply python/pyspark/pandas/utils.py:509]: Sure, let me set the default as "false"

**Comment by itholic** [comment]
On second thought, the performance benefits would not very meaningful. Rather, it may pollute the config surface so let me close it for now.

**Pull request closed without merging**